lib/pbench/test/unit/server/database/test_datasets_metadata_db.py

```diff
@@ -365,26 +365,33 @@ def test_setgetvalue_user(self, attach_dataset):
             "contact": "Wilma"
         }
 
-    @pytest.mark.parametrize("value", ["This is a name", True, 1.0, ["a", "b"]])
+    @pytest.mark.parametrize(
+        "value",
+        ["Symbols like $ and _ and @", "MY_DATA", "123456", "Shift to the left"],
+    )
     def test_mutable_dataset(self, attach_dataset, value):
         """
         Try setting a few valid names.
         """
         ds = Dataset.query(name="drb")
         Metadata.setvalue(ds, "dataset.name", value)
-        assert Metadata.getvalue(ds, "dataset")["name"] == str(value)
+        assert Metadata.getvalue(ds, "dataset")["name"] == value
 
-    def test_mutable_dataset_bad(self, attach_dataset):
+    @pytest.mark.parametrize(
+        "value",
+        ["", b"\xe8", "This is a really long name that exceeds the maximum length"],
```

Comment:
The `b"\xe8"` fails, not because of its bit pattern but because of its type (`bytes`), right?  So, I don't think this quite tests what we want it to.

Reply:
True, but as I got more into playing with test cases late Friday and this morning I've discovered that I can't actually create bad unicode strings anyway since code that's merged needs to pass basic syntax checks and an invalid u"" string or "\uxxx" code point is a Python syntax error detected by flake8. Nor can I `decode` a bad byte string here in the unit test, because the `.decode` would fail.

Right now I'm not seeing any way around this. Because utf-8 can encode any Unicode code point, the only way to provoke a failure is with an illegal Unicode code point, which I can't represent anyway.

I'm afraid I went into this thinking that UTF-8 was a subset of UTF-16, and that a "valid UTF-16" would fail the `encode`, (and that test case wasn't quite right anyway, despite the hex representation)... but that's not the case, so it really doesn't matter anyway. 😦

Reply:
It seems ironic that Python, with its "duck typing", makes it prohibitively hard to create a "bad" string (or, perhaps it's _because_ of "duck typing"...that is, it cannot be a duck if it contains un-duck-like stuff!).  (However, if it's only Flake8 which is standing in the way, here, then this seems like an appropriate spot for a localized "disable checks" annotation, if that would enable the test.)

Anyway, if you end up making another change, I think it would be appropriate to replace the `bytes` object with a `list` or `dict` or something more obviously of the "wrong type", since, as things stand, it's not the "string contents" which provoke the failure.

Reply:
Yeah, but it's not *just* `flake8`, or LGTM: it's a Python ***syntax error***, so Python blows up parsing the test file even if the `flake8` warning was suppressed.